docs/proposal-multi-user-approach.md

```diff
@@ -13,11 +12,344 @@ via a UI Server running as the suite owner for UI purposes meaning that, from
+        },                                # all privileges.
+    },
+    "<server_owner_2>": {       # For specific UI Server owner,
+        "<user2>": {            # Specific user2
+           "limit": "ALL"       # Can only be granted a maximum of ALL (READ and ADMIN) by
+        },                      # server_owner2, default access for user2 falls back to
+                                # standard READ only (if server_owner_2/user2 are
+                                # included in other auth config e.g. the top example),
+                                # or none if not in any other auth config sections.
+
+        "group:<groupA>": {                # group denoted with a `group:`
+            "default": ["READ", "ADMIN"]   # groupA has default READ, ADMIN access to server_owner_2's
+        },                                 # workflows
+    },
+    "group:<grp_of_svr_owners>":{          # Group of users who own UI Servers
+        "group: groupB": {
+            "default": "READ",             # can grant groupB users up to Admin
+            "limit": [                     # privileges, without stop and kill
+                "READ", "ADMIN", "!stop",  # operations
+                "!kill"
+            ]
+        },
+    }
+}
+```
+
+Whilst most sites and user configs will likely be simpler than this, the above config shows the flexibility available to users with the proposed system.
+
+### Site vs. User Config Precedence, Defaults and Limits
+
+At CylcCon2020 is was agreed that a:
+
+> user can ramp up authorization levels as far as the site allows
+
+Site config `limit` will determine if a user has the privileges to give away access to their workflows. It will also set a `default` access level. Users cannot raise access levels in their UI Server config for a given user or group, higher than those set in site config `limit`.
+
+If the site config does not set access for a given user or group but a UI Server owner grants access, access will be blocked.
+
+Unset defaults for both the `limit` and `default` will need consideration, I suggest that, in keeping with the deny by default principle:
+
+* if a limit is not set but a default is, then the default level is used as the limit.
+* if a default is not set but a limit is, then the default should be no access.
+
+If a user does not appear in a ui-server owners config, then the default access from site config will apply. Otherwise, defaults will not impact upon access.
+
+We will need to consider the desired behaviour if a user appears twice with different defaults and limits set, this is probably most likely to occur when a user appears in either multiple groups or in a group and as a user.
+
+Proposal: all permissions are additive, if user appears elsewhere in config, the permission level should be taken as the greatest possible. Note, default in site config does not contribute to permissions, unless user does not appear in ui server owner's config.
```

Comment:
That's sensible. If the higher level of permission is granted in the config, then no one can rightfully complain about us granting it (even if lower levels are explicitly granted as well).